### PR TITLE
Porting encoder rnn

### PR DESCRIPTION
**Pull request by GreatRSingh**
## Description

Fix #(issue)

## Type of change

Please check the option that is related to your PR.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/models/torch_models/layers.py [hunk 1]
```diff
@@ -4792,3 +4792,86 @@ def forward(self, dist: torch.Tensor) -> torch.Tensor:
...
+
+    Examples
+    --------
+    >>> from deepchem.models.torch_models.layers import EncoderRNN
+    >>> import torch 
+    >>> embedding_dimensions = 7
+    >>> num_input_token = 4
+    >>> input = torch.tensor([[1, 0, 2, 3, 0], [0, 0, 0, 0, 0], [0, 0, 0, 0, 0]])
+    >>> layer = EncoderRNN(num_input_token, embedding_dimensions)
+    >>> emb, hidden = layer(input)
```
Comment (tonydavis629) [inline deepchem/models/torch_models/layers.py:4814]: may want to show the length of the embedding is 7

**Review by tonydavis629 — APPROVED** [review]
LGTM

File: deepchem/models/torch_models/layers.py [hunk 2]
```diff
@@ -4792,3 +4792,88 @@ def forward(self, dist: torch.Tensor) -> torch.Tensor:
         dist = dist.unsqueeze(-1) / self.cutoff
         output: torch.Tensor = self.envelope(dist) * (self.freq * dist).sin()
         return output
+
+
+class EncoderRNN(nn.Module):
+    """Encoder Layer for SeqToSeq Model.
```
Comment (rbharath) [inline deepchem/models/torch_models/layers.py:4798]: Need to add this to layers.rst and torch_layers.csv

**Review by rbharath — COMMENTED** [review]
Just need to add to the docs and then good to merge

File: deepchem/models/torch_models/layers.py (hunk 1 above)
Comment (GreatRSingh) [reply deepchem/models/torch_models/layers.py:4814]: Done

File: deepchem/models/torch_models/layers.py (hunk 2 above)
Comment (GreatRSingh) [reply deepchem/models/torch_models/layers.py:4798]: Done

**Review by rbharath — APPROVED** [review]
LGTM

**Pull request merged**